src/pamiq_core/threads/base.py

```diff
@@ -0,0 +1,40 @@
+import logging
+from abc import ABC, abstractmethod
+from typing import ClassVar
+
+from pamiq_core.state_persistence import PersistentStateMixin
+from pamiq_core.threads.thread_control import ThreadEventMixin
+from pamiq_core.threads.thread_types import ThreadTypes
+from pamiq_core.utils.reflection import get_class_module_path
+
+
+class Thread(ABC, PersistentStateMixin, ThreadEventMixin):
+    """Base class for all threads.
+
+    This class provides a common interface for all threads in the system.
+
+    Attributes:
+        THREAD_TYPE (ClassVar[ThreadTypes]): The type of the thread.
```

Comment:
そうですね、二重管理になってしまうので `THREAD_TYPE`の型は docstringに記述しない方が良いですね。
あと Subclass で 定義が必要なことも示すと良いです。
```suggestion
    Attributes:
        THREAD_TYPE: The type of the thread. Subclasses must define this class variable.
```

Reply:
ありがとうございます、承知しました！